src/main/java/com/cloudbees/hudson/plugins/folder/computed/PeriodicFolderTrigger.java

```diff
@@ -159,14 +171,24 @@ public String getInterval() {
      */
     @Override
     public void run() {
+        long now = System.currentTimeMillis();
         FolderComputation<?> computation = job.getComputation();
         if (computation != null) {
-            long delay = System.currentTimeMillis() - computation.getTimestamp().getTimeInMillis();
+            long delay = now - computation.getTimestamp().getTimeInMillis();
             if (delay < interval) {
+                LOGGER.log(Level.FINE, "Too early to reschedule {0} based on last computation", job);
                 return;
             }
         }
-        job.scheduleBuild(0, new TimerTrigger.TimerTriggerCause());
+        if (now - lastTriggered < interval) {
+            LOGGER.log(Level.FINE, "Too early to reschedule {0} based on last triggering", job);
+            return;
+        }
+        if (job.scheduleBuild(0, new TimerTrigger.TimerTriggerCause())) {
+            lastTriggered = now;
```

Comment:
In retrospect this feels like a hack. Maybe it would suffice to just pass a delay of a few seconds to `scheduleBuild`, so that if cron runs this trigger several times in close succession they would all be coalesced?